File: src/main/java/com/mestrap/entity/Task.java
package com.mestrap.entity;

import java.util.List;

/**
 * @author melvek
 */
public class Task {
    private String name;
    private String description;
    private List<Step> steps;

    public String getName() { return name; }
    public void setName(String name) { this.name = name; }

    public String getDescription() { return description; }
    public void setDescription(String description) { this.description = description; }

    public List<Step> getSteps() { return steps; }
    public void setSteps(List<Step> steps) { this.steps = steps; }
}

File: internal/config/task.go
package config

// Task 是一个任务，由若干步骤组成。
type Task struct {
	Description string `yaml:"description,omitempty"`
	Steps       []Step `yaml:"steps"`
}

// Step 是任务中的一步，引用一个 Action。
type Step struct {
	Name   string                 `yaml:"name"`
	Action string                 `yaml:"action"`
	With   map[string]interface{} `yaml:"with"`
	Delay  int                    `yaml:"delay"`
}
